### PR TITLE
Task 11-Sort articles

### PR DESCRIPTION
**Pull request by jishajohn88**

### CODE REVIEW

File: api.js [hunk 1]
```diff
@@ -7,19 +7,26 @@ const api = axios.create({
 const getArticlesByCreateDate = () => {
   return api.get("articles?sort_by=created_at").then(({ data }) => {
     return data.articles;
+  }).catch((err) => {
+    return err
   });
 };
 
+const getArticles = (sortBy="created_at",order="asc") =>{
```
Comment (nc-saima-khanom) [inline api.js:15]: As your article cards don't currently display any date, vote or comment count, it is difficult for the user to tell if the articles have actually been sorted by what they have selected - they can only tell that something has happened as the page has refreshed and there may be a different article at the top. I'd recommend adding some more info to your article cards to show this clearly

File: api.js [hunk 2]
```diff
@@ -7,19 +7,26 @@ const api = axios.create({
 const getArticlesByCreateDate = () => {
   return api.get("articles?sort_by=created_at").then(({ data }) => {
     return data.articles;
+  }).catch((err) => {
```
Comment (nc-saima-khanom) [inline api.js:10]: we shouldn't have catch blocks in our request functions as the error should be dealt with inside the actual component it is being used - this catch block is not adding any value as it's just returning the err which already happens by default when the promise rejects

File: api.js [hunk 3]
```diff
@@ -7,19 +7,26 @@ const api = axios.create({
 const getArticlesByCreateDate = () => {
```
Comment (nc-saima-khanom) [inline api.js:7]: do you need this function if the next one has a default sort by param for created at?

File: src/components/Footer.jsx [hunk 4]
```diff
@@ -1,9 +1,43 @@
...
\ No newline at end of file
+const Footer = (props) => {
+
+    const {articles} = props
+    const [page,setPage] = useState(1)
+    const pages = Math.ceil(articles.total_count / 10)
+    const [articleCount,setArticlesCount] = useState([])
+    const [totalcount,setTotalCount] = useState(articles.total_count)
+    const [error,setError] = useState(null)
+    useEffect(() => {
```
Comment (nc-saima-khanom) [inline src/components/Footer.jsx:13]: We should also have a loading state for the articles here

**Comment by nc-saima-khanom** [comment]
Looks great so far, well done! :D

File: api.js (hunk 1 above)
Comment (jishajohn88) [reply api.js:15]: @nc-saima-khanom do you mean I add it to the card on the home page with additional details?

Comment (nc-saima-khanom) [reply api.js:15]: Yes exactly :)

**Pull request merged**